src/combine.ts

```diff
@@ -28,8 +29,18 @@ export function combineWith<T extends [...Producer<unknown>[]], U>(
           if (++numStarted === 1) sink(0, talkback);
         } else if (t === 1) {
           combined[i] = d;
-          if (combined.indexOf(undefined) === -1)
+          if (!filled) {
+            filled = true;
+            for (let i = 0; i < combined.length; i++) {
+              if (!(i in combined)) {
+                filled = false;
+                break;
+              }
```

Comment:
This for loop (which will run many times) smells to me much slower than a counter variable solution.

Check the callbag-combine implementation. Apart from the obscurely named variables, the implementation itself should be easy to build.

Reply:
the trick with the empty object is a good one. Did that now